src/arfsdao.ts

```diff
@@ -709,6 +709,21 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 		return latestRevisionsOnly ? allFiles.filter(latestRevisionFilter) : allFiles;
 	}
 
+	async getChildrenFolderIds(
+		folderId: FolderID,
+		allFolderEntitiesOfDrive: ArFSPublicFolder[] | ArFSPrivateFolder[]
+	): Promise<FolderID[]> {
+		const hierarchy = FolderHierarchy.newFromEntities(allFolderEntitiesOfDrive);
+		return hierarchy.folderIdSubtreeFromFolderId(folderId, Number.MAX_SAFE_INTEGER);
+	}
+
+	async getPrivateChildrenFolderIds(folderId: FolderID, driveId: DriveID, driveKey: DriveKey): Promise<FolderID[]> {
```

Comment:
Consider deconstructing params from a type to avoid using FolderID and DriveID in the wrong positions.